ui/src/app/shared/utils/url-search-params.utils.ts

```diff
@@ -111,6 +112,31 @@ import {FieldDataType, queryDataTypes, queryExtensionsDataTypes} from "../common
     return queryRequest;
   }
 
+  /** Get the start time by the given time frame on the dashboard page */
+  public static getStartTimeByTimeFrame(timeFrame: TimeFrame): Date {
+    const now = new Date;
+    switch (timeFrame) {
+      case TimeFrame.HOURS1:
+        now.setHours(now.getHours() - 1);
+        break;
+      case TimeFrame.HOURS8:
+        now.setHours(now.getHours() - 24);
```

Comment:
I don't think this is going to work if hours <= 8

Reply:
It dose work. I tested it: now is UTC 8/8 22:00, I subtract by 23 and it gives 8/7 23:00.